### PR TITLE
Feature/issue3 markdown to partitions

### PR DESCRIPTION
**Pull request by Pergamene**
Not as neat as I'd like but it's functional and ready to be added to the site.

### CODE REVIEW

**Comment by rhyeen** [comment]
Resolve them merge conflicts!

File: src/markdown-combination.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  }
+]
+
+for (let testCase of testCases) {
+  test(testCase.name, () => {
+    debugger;
```
Comment (rhyeen) [inline src/markdown-combination.test.js]: delete

File: src/markdown-to-inner-partitions.js [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    return { type: PARTITION_TYPES.BOLD, value: markdownSubstring.replace(/\\/g, '') };
+  } else {
+    return { type: PARTITION_TYPES.BOLD, partitions };
+  }
+}
+
+function _buildItalicsPartition(markdownSubstring) {
+  let partitions = generateInnerPartitions(markdownSubstring);
+  if (!partitions) {
+    return { type: PARTITION_TYPES.ITALICS, value: markdownSubstring.replace(/\\/g, '') };
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.js]: You use `markdownSubstring.replace(/\\/g, '')` a lot.  Why not make it a helper function?  That way it'll make more sense anyway.  Like `removeEscapedDelimiter` or something.

File: src/markdown-to-inner-partitions.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+function _buildColorPartition(breaks) {
+  return { type: PARTITION_TYPES.COLOR, value: breaks[0], color: breaks[1] };
+}
+
+function _buildTextPartition(markdownSubstring) {
+  return { type: PARTITION_TYPES.TEXT, value: markdownSubstring.replace(/\\/g, '') };
+}
+
+// HELPER METHODS
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.js]: Aren't all the `_*` (private) functions helper functions?

File: src/markdown-to-inner-partitions.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  // {
+  //   name: 'test for bold',
+  //   paramMarkdown: '*This is a test for bold partitions*',
+  //   returnPartitions: [{type: 'bold', value: 'This is a test for bold partitions'}]
+  // },
+  // {
+  //   name: 'test for italics',
+  //   paramMarkdown: '_This is a test for italics partitions_',
+  //   returnPartitions: [{type: 'italics', value: 'This is a test for italics partitions'}]
+  // },
```
Comment (rhyeen) [inline src/markdown-to-inner-partitions.test.js]: Are these two zombie code or should they be uncommented?

File: src/markdown-to-partitions.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
+import { PARTITION_TYPES } from './partition-types';
+import { generateInnerPartitions } from './markdown-to-inner-partitions';
+
+/**
+ * assemble all partition fragments
+ * 
+ * @param {string} markdownText see partitions-to-markdown 
+ */
+export function generatePartitions(markdownText) {
+  markdownText = markdownText.replace(/(\n+\s*\n*)/g, '\n');
```
Comment (rhyeen) [inline src/markdown-to-partitions.js]: Make helper function such as `conformNewlines`

File: src/markdown-to-partitions.js [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+function _buildParagraphPartition(markdownSubstring) {
+  let partitions = generateInnerPartitions(markdownSubstring);
+  if (!partitions) {
+    partitions = [{ type: PARTITION_TYPES.TEXT, value: markdownSubstring.replace(/\\/g, '') }];
+  }
+  return { type: PARTITION_TYPES.PARAGRAPH, partitions };
+}
+
+// HELPER METHODS
```
Comment (rhyeen) [inline src/markdown-to-partitions.js]: same comment here

File: src/markdown-to-inner-partitions.js (hunk 3 above)
Comment (Pergamene) [reply src/markdown-to-inner-partitions.js]: Yes they are but the functions about the helper comment are for a specific partition. The functions below that are things that do the general (non-partition-specific) work. It made sense to me.

File: src/markdown-to-inner-partitions.test.js (hunk 4 above)
Comment (Pergamene) [reply src/markdown-to-inner-partitions.test.js]: Oh I commented them for debugging they should be uncommented.

**Pull request merged**